test/e2e/util.go

```diff
@@ -679,7 +680,13 @@ func waitForNamespacesDeleted(c *client.Client, namespaces []string, timeout tim
 }
 
 func waitForServiceAccountInNamespace(c *client.Client, ns, serviceAccountName string, timeout time.Duration) error {
-	Logf("Waiting up to %v for service account %s to be provisioned in ns %s", timeout, serviceAccountName, ns)
+	w, err := c.ServiceAccounts(ns).Watch(api.SingleObject(api.ObjectMeta{Name: serviceAccountName}))
```

Comment:
Thinking about this more, I still think you need a function that implements this patter, but with a retry loop around it. In case the watch drops due to network etc. I think this will cause rare flakes otherwise.